torch/distributed/tensor/parallel/style.py

```diff
@@ -1,652 +1,357 @@
-    output: DTensor, _device_mesh: Optional[DeviceMesh] = None
-) -> torch.Tensor:  # noqa: D205, D400
-    """
-    .. warning::
-        This method was deprecated and please specify ``output_layouts`` instead.
-    """
-    _deprecate_warnings("make_sharded_output_tensor", "Specify output_layouts instead.")
-    return output.to_local()  # type: ignore[call-arg]
+    def apply(self, module: nn.Module, device_mesh: DeviceMesh) -> nn.Module:
+        ...
 
 
-@_prepare_output_validate  # type: ignore[arg-type] # pyre-ignore[56]
-def make_output_reshard_tensor(
-    output: DTensor,
-    device_mesh: Optional[DeviceMesh] = None,
-) -> torch.Tensor:  # noqa: D205, D400
-    """
-    .. warning::
-        This method was deprecated and please specify ``output_layouts`` instead.
-    """
-    _deprecate_warnings("make_output_reshard_tensor", "Specify output_layouts instead.")
-    return make_output_shard_1d(output, device_mesh).to_local()  # type: ignore[call-arg, attr-defined, misc]
-
-
-def _redistribute_per_both_layouts(t, input_layout, output_layout, device_mesh):
-    dst_placements = (output_layout,)
-    if isinstance(t, DTensor):
-        return t.redistribute(device_mesh, dst_placements)
-    elif isinstance(t, torch.Tensor):
-        dtensor = DTensor.from_local(t, device_mesh, (input_layout,), run_check=False)
-        return dtensor.redistribute(device_mesh, dst_placements)
-    else:
-        if input_layout is not None:
-            raise RuntimeError(
-                "Tensor parallel module expects DTensor or tensor"
-                f" when layout specified but received {type(t)}!"
-            )
-        else:
-            return t
-
-
-def _redistribute_per_layout(layout, use_local_output, t, device_mesh):
-    dst_placements = (layout,)
-    if isinstance(t, DTensor):
-        dtensor = t.redistribute(device_mesh, dst_placements)
-        return dtensor.to_local() if use_local_output else dtensor
-    else:
-        if layout is not None:
-            raise RuntimeError(
-                "Tensor parallel module expects DTensor or tensor"
-                f" when layout specified but received {type(t)}!"
-            )
-        else:
-            return t
-
-
-class RowwiseParallel(ParallelStyle):
+class ColwiseParallel(ParallelStyle):
     """
-    Partition the row of a module.
-
-    We assume the input to be a sharded :class:`DTensor` and output to be a :class:`torch.Tensor`.
-
-    Args:
-        input_layouts (Union[Placement, Tuple[Placement, ...]]):
-            The layout of input tensor(s) which DTensor will be created upon.
-        output_layouts (Union[Placement, Tuple[Placement, ...]]):
-            The layout of input tensor(s) which created DTensor will be redistributed to.
-        use_local_output (bool):
-            Whether to convert the DTensor to local :class:`torch.Tensor`.
-
+    Partition the column of a nn.Module and allow the parallelized module to run colwise sharded computation.
```

Comment:
nit: the phrasing doesn't make sense to me?  ("column of a nn.module") 

I guess you mean "partition an nn.Module in a column-wise fashion"?  (that's more clear but still could use more defining what it means or if it assumes the nn.Module is of a certain shape/format.  (must be a single Linear?)